akka-stream/src/main/scala/akka/stream/javadsl/Hub.scala

```diff
@@ -91,3 +94,122 @@ object BroadcastHub {
+ * actually live consumers it has. The selection of consumer is done with a function. Each element can be routed to
+ * only one consumer. Once the producer has been materialized, the [[Sink]] it feeds into returns a
+ * materialized value which is the corresponding [[Source]]. This [[Source]] can be materialized an arbitrary number
+ * of times, where each of the new materializations will receive their elements from the original [[Sink]].
+ */
+object PartitionHub {
+
+  /**
+   * Creates a [[Sink]] that receives elements from its upstream producer and routes them to a dynamic set
+   * of consumers. After the [[Sink]] returned by this method is materialized, it returns a [[Source]] as materialized
+   * value. This [[Source]] can be materialized an arbitrary number of times and each materialization will receive the
+   * elements from the original [[Sink]].
+   *
+   * Every new materialization of the [[Sink]] results in a new, independent hub, which materializes to its own
+   * [[Source]] for consuming the [[Sink]] of that materialization.
+   *
+   * If the original [[Sink]] is failed, then the failure is immediately propagated to all of its materialized
+   * [[Source]]s (possibly jumping over already buffered elements). If the original [[Sink]] is completed, then
+   * all corresponding [[Source]]s are completed. Both failure and normal completion is "remembered" and later
+   * materializations of the [[Source]] will see the same (failure or completion) state. [[Source]]s that are
+   * cancelled are simply removed from the dynamic set of consumers.
+   *
+   * This `statefulSink` should be used when there is a need to keep mutable state in the partition function,
+   * e.g. for implemening round-robin or sticky session kind of routing. If state is not needed the [[#of]] can
+   * be more convenient to use.
+   *
+   * @param partitioner Function that decides where to route an element. It is a factory of a function to
+   *   to be able to hold stateful variables that are unique for each materialization. The function
+   *   takes two parameters; the first is information about active consumers, including an array of consumer
+   *   identifiers and the second is the stream element. The function should return the selected consumer
+   *   identifier for the given element. The function will never be called when there are no active consumers,
+   *   i.e. there is always at least one element in the array of identifiers.
+   * @param startAfterNbrOfConsumers Elements are buffered until this number of consumers have been connected.
+   *   This is only used initially when the stage is starting up, i.e. it is not honored when consumers have
+   *   been removed (canceled).
+   * @param bufferSize Total number of elements that can be buffered. If this buffer is full, the producer
+   *   is backpressured.
+   */
+  def ofStateful[T](clazz: Class[T], partitioner: Supplier[BiFunction[ConsumerInfo, T, java.lang.Long]],
```

Comment:
Yeah, there though it did only have the `of` not a stateful one... I guess it does not matter that much, the meaning makes sense